File: app/src/main/java/com/rowland/delivery/merchant/application/di/components/ApplicationComponent.java
package com.rowland.delivery.merchant.application.di.components;

import android.content.Context;

import com.rowland.delivery.merchant.application.di.modules.ContextModule;

import dagger.Component;

/**
 * Created by Rowland on 4/30/2018.
 */

@Component(modules = ContextModule.class)
public interface ApplicationComponent {

    Context getContext();
}

File: app/src/main/java/com/rowland/delivery/merchant/application/di/components/ApplicationComponent.kt
package com.rowland.delivery.merchant.application.di.components

import android.content.Context

import com.rowland.delivery.merchant.application.di.modules.ContextModule

import dagger.Component

/**
 * Created by Rowland on 4/30/2018.
 */

@Component(modules = arrayOf(ContextModule::class))
interface ApplicationComponent {

    val context: Context
}
